therapy/etl/drugsatfda.py

```diff
@@ -0,0 +1,157 @@
+            "None (Tentative Approval)": ApprovalStatus.FDA_TENTATIVE.value,
+        }
+        statuses = [p["marketing_status"] for p in products]
+        if not all([s == statuses[0] for s in statuses]):
+            msg = (f"Application {concept_id} has inconsistent marketing "
+                   f"statuses: {statuses}")
+            logger.info(msg)
+            return None
+        else:
+            return statuses_map.get(statuses[0])
+
+    def _transform_data(self) -> None:
+        """Prepare source data for loading into DB."""
+        with open(self._src_file, "r") as f:
+            data = json.load(f)["results"]
+
+        for result in data:
+            if "products" not in result:
+                continue
+            products = result["products"]
+
+            app_no = result["application_number"]
+            if app_no.startswith("BLA"):
+                # ignore biologics license applications (tentative)
+                continue
+            concept_id = f"{NamespacePrefix.DRUGSATFDA.value}:{app_no}"
+            therapy: RecordParams = {"concept_id": concept_id}
+
+            status = self._get_marketing_status(products, concept_id)
+            if status:
+                therapy["approval_status"] = status
+
+            brand_names = [p["brand_name"] for p in products]
+
+            aliases = []
+            if "openfda" in result:
+                openfda = result["openfda"]
+                brand_name = openfda.get("brand_name")
+                if brand_name:
+                    brand_names += brand_name
+
+                substances = openfda.get("substance_name", [])
+                n_substances = len(set(substances))
+                if n_substances > 1:
+                    # if ambiguous, store all as aliases
+                    msg = (f"Application {concept_id} has {n_substances} "
+                           f"substance names: {substances}")
+                    logger.debug(msg)
+                    aliases += substances
+                elif n_substances == 1:
+                    therapy["label"] = substances[0]
+
+                generics = openfda.get("generic_name", [])
+                n_generic = len(set(generics))
+                if n_generic > 1:
+                    aliases += generics
+                elif n_generic == 1:
+                    if n_substances == 0:
+                        # there are about 300 cases of this
+                        therapy["label"] = generics[0]
+                    else:
+                        aliases.append(generics[0])
+
+                unii = openfda.get("unii")
+                if unii:
+                    therapy["associated_with"] = [f"{NamespacePrefix.UNII.value}:{u}"
+                                                  for u in unii]
+                rxcui = openfda.get("rxcui")
+                if rxcui:
+                    therapy["xrefs"] = [f"{NamespacePrefix.RXNORM.value}:{r}"
+                                        for r in rxcui]
+
```

Comment:
Should we be adding `product_ndc` ( [National Drug Codes List](https://ndclist.com) ) or `spl_id` (https://open.fda.gov/data/spl/) to `associated_with`?

Reply:
^^ definitely at least `product_ndc` since it's listed on identifiers.org -- question for @mcannon068nw: either/both?

Reply:
I would say keep both. Though we are not doing anything with it ourselves right now, having the spl_id to link to could potentially be useful for someone who wants to easily find the product label for indication, reactions, etc.